v-next/hardhat-node-test-runner/src/task-action.ts

```diff
@@ -103,6 +131,17 @@ const testWithHardhat: NewTaskActionFunction<TestActionArguments> = async (
 
     await pipeline(reporterStream, createNonClosingWriter(process.stdout));
 
+    if (hre.globalOptions.coverage === true) {
+      assertHardhatInvariant(
+        process.env.HARDHAT_NODE_TEST_COVERAGE_PATH !== undefined,
+        "HARDHAT_NODE_TEST_COVERAGE_PATH should be defined",
+      );
+      const report: CoverageReport = await readJsonFile(
+        process.env.HARDHAT_NODE_TEST_COVERAGE_PATH,
+      );
+      log("Coverage report", report);
```

Comment:
Please note the display of the report as well as bubbling of the data up the stack will be handled separately.